### ISSUE TITLE
error if I enable "Track Combat Move History" option

### ISSUE THREAD

**Issue by CaosFR** [caewok/fvtt-elevation-ruler#201]
Hi,

After install elevation-ruler I have this error if I enable "Track Combat Move History" :

Ruler.js:443 Uncaught TypeError: undefined. segment.ray.B.clone is not a function
[Detected 2 packages: elevationruler(0.10.9), lib-wrapper(1.12.15.0)]
    at Ruler._computeDistance (Ruler.js:443:29)
    at 🎁CONFIG.Canvas.rulerClass.prototype._computeDistance#0 (libWrapper-wrapper.js:187:52)
    at Ruler.measure (foundry.js:36092:10)
    at Ruler._onMouseMove (foundry.js:36710:10)
    at #onDragLeftMove (foundry.js:25902:67)
    at MouseInteractionManager.callback (foundry.js:30823:17)
    at #handleDragMove (foundry.js:31202:26)
    at #handlePointerMove (foundry.js:31167:64)
    at vh.notifyListeners (EventBoundary.ts:1464:33)
    at vh.notifyTarget (EventBoundary.ts:658:14)
    at vh.propagate (EventBoundary.ts:310:18)
    at vh.dispatchEvent (EventBoundary.ts:210:14)
    at vh.mapPointerMove (EventBoundary.ts:815:64)
    at vh.mapEvent (EventBoundary.ts:231:28)
    at ra.onPointerMove (EventSystem.ts:396:31)

No other module are enable

Thanks

**Comment by WizzardMaker** [comment]
Happens to me aswell

I noticed, that a segment in the path seems to only be of type `Ray` instead of `Ray3D` with segment.ray.B just being an object not a class with the clone function

An excerpt of the values:
```
[
    {
        "A": {
            "x": 1550,
            "y": 1250,
            "z": 0
        },
        "B": {
            "x": 1350,
            "y": 1050,
            "z": 0
        },
        "y0": 1250,
        "x0": 1550,
        "dx": -200,
        "dy": -200,
        "slope": 1,
        "dz": 0
    },
    {
        "A": {
            "x": 1350,
            "y": 1050,
            "z": 0
        },
        "B": {
            "x": 1150,
            "y": 850,
            "z": 0
        },
        "y0": 1050,
        "x0": 1350,
        "dx": -200,
        "dy": -200,
        "slope": 1,
        "dz": 0
    },
    {
        "A": {
            "x": 1150,
            "y": 850,
            "z": 0
        },
        "B": {
            "x": 1450,
            "y": 1250,
            "z": 0
        },
        "y0": 850,
        "x0": 1150,
        "dx": 300,
        "dy": 400,
        "slope": 1.3333333333333333,
        "dz": 0
    },
    {
        "_distance": 0,
        "A": {
            "x": 1450,
            "y": 1250,
            "teleport": false,
            "cost": 0,
            "z": 0
        },
        "B": {
            "x": 1450,
            "y": 1250,
            "_userElevationIncrements": 0,
            "_forceToGround": false,
            "elevation": 0
        },
        "y0": 1250,
        "x0": 1450,
        "dx": 0,
        "dy": 0,
        "slope": null
    },
    {
        "A": {
            "x": 1450,
            "y": 1250,
            "z": 0
        },
        "B": {
            "x": 1450,
            "y": 1250,
            "z": 0
        },
        "y0": 1250,
        "x0": 1450,
        "dx": 0,
        "dy": 0,
        "slope": null,
        "dz": 0
    }
]
```
![image](https://github.com/user-attachments/assets/a4f21c51-2bc8-492e-bb88-0d4149f649f6)

**Comment by caewok** [comment]
Thanks! I addressed this in v0.10.10.
